state/pg_state_manager.go

```diff
@@ -125,8 +126,20 @@ func (sm *SQLStateManager) Initialize(conf config.Config) error {
 		return errors.Wrap(err, "unable to open postgres db")
 	}
 
+	var readonlyDbUrl string
+	if conf.IsSet("readonly_database_url") {
+		readonlyDbUrl = conf.GetString("readonly_database_url")
+	} else {
+		readonlyDbUrl = dburl
```

Comment:
Should we fallback to the main DB? Maybe add a warning/log here to explicitly state that we're using the main DB since a readonly wasn't specified.

Reply:
good point

Reply:
yeah I didn't want to make it a hard requirement to have a read replica.